mule-module-apikit/src/main/java/org/mule/module/apikit/validation/RestJsonSchemaValidator.java

```diff
@@ -97,11 +105,22 @@ else if (input instanceof byte[])
 
             JsonSchema schema = JsonSchemaCache.getJsonSchemaCache(muleContext, configId, api).get(schemaPath);
             ProcessingReport report = schema.validate(data);
-            if (!report.isSuccess())
+            Iterator<ProcessingMessage> iterator = report.iterator();
+
+            while (iterator.hasNext())
             {
-                String message = report.iterator().hasNext() ? report.iterator().next().toString() : "no message";
-                logger.info("Schema validation failed: " + message);
-                throw new BadRequestException(message);
+                ProcessingMessage next = iterator.next();
+                LogLevel logLevel = next.getLogLevel();
+                String message = next.toString();
+
+                boolean failOnWarning = Boolean.valueOf(
+                        System.getProperty(JSON_SCHEMA_FAIL_ON_WARNING_KEY, "false"));
+
+                if (logLevel.equals(ERROR) || (logLevel.equals(WARNING) && failOnWarning))
+                {
+                    logger.info("Schema validation failed: " + message);
+                    throw new BadRequestException(message);
```

Comment:
Let's handle this improvement in a separate issue as it can break the parsing of the error messages by the users